### PR TITLE
UPE - Add empty state when user removes all payment methods from checkout

### LINKED ISSUE #2560: Add empty state when user removes all payment methods from checkout
Hi folks!

In UPE, when merchant removes all payment methods from checkout (which could be the case if they want to use a different payment method alongside WCP), we could display an empty state picture below ([view in Figma](https://www.figma.com/file/Li3cAa1Y2uSklkek0axmyu/Local-payment-methods?node-id=4575%3A202591)).

Clicking the "Add payment method" would launch the modal.

image

### PR DESCRIPTION
**Pull request by mdmoore**
Fixes #2560 

#### Changes proposed in this Pull Request

Currently, UPE must have one payment method enabled. This allows all UPE payment methods to be deleted and displays a starting point when no payment methods have been enabled. We've received some requests to allow all UPE payment methods to be removed, allow users to accept only Apple/Google Pay. This takes care of that.

Screen Shot on 2021-11-16 at 17-34-27

#### Testing instructions

1. Navigate to **Payments > Settings** and scroll to the **Payment Methods** section
2. Delete all payment methods
3. Save the settings
4. Try to add new payment methods

- [x] Added changelog entry (or does not apply)
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

**Comment by tommyshellberg** [comment]
Looks like [JS tests are failing](https://github.com/Automattic/woocommerce-payments/runs/4231503185?check_suite_focus=true), are they failing for you locally too?

**Comment by mdmoore** [comment]
They are indeed. Looking into it now.

**Comment by mdmoore** [comment]
@tommyshellberg - Looking better now. Some tests and logic in the component needed to be refactored to accommodate the new view.

File: changelog.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 *** WooCommerce Payments Changelog ***
 
 = 3.3.0 - 2021-xx-xx =
+* Add - Add empty state when all payment methods from checkout.
```
Comment (tommyshellberg) [inline changelog.txt:4]: Missing a word, maybe? "when all payment methods are removed from checkout."?

File: client/payment-methods/test/index.js [hunk 2]
```diff
@@ -73,6 +74,22 @@ describe( 'PaymentMethods', () => {
 		expect( addPaymentMethodButton ).not.toBeInTheDocument();
 	} );
 
+	test( 'renders the "Add payment method" button when there are 0 payment methods enabled but at least one available', () => {
```
Comment (tommyshellberg) [inline client/payment-methods/test/index.js:77]: Thanks for adding a test 💪

**Comment by tommyshellberg** [comment]
> @tommyshellberg - Looking better now. Some tests and logic in the component needed to be refactored to accommodate the new view.

Thanks for fixing that! I think there's an issue on my end in that the 'Add Payment Method' button causes the app to crash:

`Uncaught TypeError: paymentMethodStatuses[wcpay_payment_methods_map__WEBPACK_IMPORTED_MODULE_14__.default[method].stripe_key] is undefined`

**Comment by mdmoore** [comment]
It looks like #3300 nullifies this PR as all payment methods are always displayed now.

**Pull request closed without merging**